Send demonstration example samples upon install

tombruijn opened the pull request:
Appsignal::Demo will send two samples for demonstration purposes. This
prevents us from asking the user to click through their site and somehow
trigger an error or a slow request. We can just as well send samples
ourselves.

This will allow us to send samples used for demonstration purposes to
the application of user once the installation is completed.

---

It will create these samples in the `development` environment of the chosen application name during installation. In the screenshots I used `Demo test` but for the user it will be their own application name.

Partially closes issue https://github.com/appsignal/appsignal-server/issues/1683
The partially refers to the UI changes we discuss below.

---

## Screenshots

Error sample

![image](https://cloud.githubusercontent.com/assets/282402/19562173/a67ab8c6-96db-11e6-96d3-0d6b4905c169.png)

Performance sample

![image](https://cloud.githubusercontent.com/assets/282402/19562157/99f55b88-96db-11e6-8377-4d27dc3fbeac.png)

wesoudshoorn commented:
Oeh nice!

- We should probably update the instructions in the flow on `appsignal-server` too, it says there as well to click around
- Can we automatically set these errors to closed? It will be weird if they are open...

tombruijn commented:
> Can we automatically set these errors to closed?

That would have to happen on the server side. We could mark the demo requests with some magic metadata. Would the server processor be able to pick that up @matsimitsu ?

matsimitsu commented:
It's possible, but why not let the user do it? That way they'll learn how to close issues and what for consequences it has :) (e.g. fill metadata fields with text on how to close an issue).

wesoudshoorn commented:
@matsimitsu that sounds like an amazing idea :)

tombruijn commented:
I noticed that the backtrace is missing from the error. Will look into that.

I also added screenshots.

And I explained in the description that these samples are added to the `development` environment of the application name chosen by the user during installation.

tombruijn commented:
> It's possible, but why not let the user do it? That way they'll learn how to close issues and what for consequences it has :) (e.g. fill metadata fields with text on how to close an issue).

@matsimitsu do you mean adding something this?

```ruby
transaction.set_metadata("close this issue", "Close this issue by clicking on the issue state in the UI.")
```

will look like this:

![image](https://cloud.githubusercontent.com/assets/282402/19592347/9615294e-977b-11e6-9375-27fa6c393a5a.png)

Not exactly the most readable, but could work for now. I'd maybe rather see some kind of extra help UI when we detect a demonstration sample (detected by this kind of metadata).

matsimitsu commented:
>  I'd maybe rather see some kind of extra help UI when we detect a demonstration sample (detected by this kind of metadata).

You could set a metadata field that says `demo_sample` or something and pick that up in the sample show. That's also fine by me :)

tombruijn commented:
I think something like the following screenshot could work. Something like an integrated tour block that only shows up for these kinds of examples. It helps the user out by guiding them through the page and some of the elements on it. Maybe some special style like a different color would help differentiate between the blocks on the page would help? I can also put it at the top of the page, but I thought the error message would be more important?

![image](https://cloud.githubusercontent.com/assets/282402/19599179/c057ff96-979f-11e6-92af-506933ee641c.png)

tombruijn commented:
Regardless of my previous comment. This PR is now ready for review @jeffkreeftmeijer @thijsc 

(I rebased and added metadata `demo_sample: true` to the sample.)

matsimitsu commented:
Nice! Let me know if you need help implementing it in React :)
